AfrikaBurn/BurnElementsViewController.swift

```diff
@@ -218,6 +259,35 @@ class BurnElementsViewModel {
         return elements.results[index]
     }
     
+    private func applySearchTextFilter(searchText : String) -> CurrentElements{
+        
+        let newElements: CurrentElements
+        
+        // if it's an empty search then just show previous filter
+        if (searchText.isEmpty){
+            switch displayMode {
+            case .default:
+                newElements = .normal(applyFilter(activeFilter, to: self.allElements))
+            case .favorites:
+                newElements = .favorites(applyFilter(activeFilter, to: persistentStore.favorites()))
+            }
+            return newElements
+        }
+        
+        
+        let searchedElements = self.allElements.filter("name CONTAINS[c] '\(searchText)' OR longBlurb CONTAINS[c] '\(searchText)'")
```

Comment:
One thing you can do with realm is chain filters. I think for this instance what we should do is maybe update the `Filter` object to include the search text as an optional string. that way you can keep a search active when switching between all and favorites where the current solution doesn't allow for this.

So if we have

```swift
private func applyFilter<T: AfrikaBurnElement>(_ filter: Filter, to elements: Results<T>) -> Results<T> {
    var result: Results<T>
    if let elementType = activeFilter.elementType {
        result = elements.filter(type: elementType)
    } else {
        result = elements
    }
    
    if let searchText = filter.searchText {
        result = result.filter("name CONTAINS[c] '\(searchText)' OR longBlurb CONTAINS[c] '\(searchText)'")
    }  
    return result
}
```
This gives our filter function more power since Filter has expanded and now we still use the same mechanism to filter both by type and with search text.

What do you think?